packages/features/bookings/lib/getBookingFields.ts

```diff
@@ -267,6 +267,28 @@ export const ensureBookingInputsHaveSystemFields = ({
         },
       ],
     },
+    {
+      defaultLabel: "cancellation_reason",
+      type: "textarea",
+      editable: "system-but-optional",
+      name: "cancelReason",
+      defaultPlaceholder: "cancellation_reason_placeholder",
+      required: false,
+      hidden: false,
+      views: [
+        {
+          id: "cancellationReason",
+          label: "Cancel View",
+        },
+      ],
+      sources: [
+        {
+          label: "Default",
+          id: "default",
+          type: "default",
+        },
+      ],
+    },
```

Comment:
@Chiranjeev-droid thanks for the work. It works, but the cancellation reason is not part of the booking fields. I’m not sure we can consider this a viable fix

Reply:
if needed , we can create a field in database saying cancellationReasonNeeded inside the eventtype and just add a toogle on the advanced tab. Based on the boolean value , we can make the reason required or optional but then user wont be able to customize the cancellationreason field.

Reply:
Yeah, we got approval from the team. Let’s make it a setting toggle in the advanced tab instead of a booking question